### ISSUE TITLE
Question about training parameters

### ISSUE THREAD

**Issue by zhifeis** [jhb86253817/PIPNet#32]
I'm trying to use the PIPNet framework to train a custom model unrelated to human faces with a non-standard number of keypoints (4). I'm confused about this line in lib/train.py:

`labels = get_label(cfg.data_name, 'train.txt')`

what is labels, what format is it and what is train.txt?

**Comment by jhb86253817** [comment]
Hi,

Both 'meanface_indices' and 'num_nb' are related to the neighbor regression
module described in the paper, which predicts the neighbors of each
landmark and average the predictions to make the final result robust.
'meanface_indices' stores the the information that which landmarks are
neighbors of which; 'num_nb' is a hyperparameter for the number of
neighbors each landmark predict.

To adopt it to your task, you would need to modify more than just
'num_lms'. First of all, you need to have a file stores mean shape of the
four landmarks, similar to 'data/data_300W/meanface.txt'. Actually, you can
simply remove the neighbor regression module for simplicity, but it also
requires some modification of the codes.
Secondly, you need to specify the points_flip information to tell the model
if image is horizontally flipped, how the landmark would change. But if
your four landmarks are not symmetry, then you just need to remove the
horizontal flip data augmentation.

Anyway, the adoptation would requires some code works. You may try to
understand the whole framework first.


zhifeis ***@***.***> 于2022年9月21日周三 23:33写道：
